src/super_gradients/training/utils/optimizer_utils.py

```diff
@@ -117,6 +126,97 @@ def build_optimizer(net: nn.Module, lr: float, training_params) -> optim.Optimiz
         optimizer_training_params = net_named_params
 
     # CREATE AN OPTIMIZER OBJECT AND INITIALIZE IT
-    optimizer = optimizer_cls(optimizer_training_params, lr=lr, **training_params.optimizer_params)
+    optimizer = optimizer_cls(optimizer_training_params, **training_params.optimizer_params)
+
+    return optimizer
+
+
+def separate_lr_groups(model: nn.Module, lr_dict: Dict[str, float]) -> List[Dict]:
+    """
+    Separate parameters based on specified learning rates for each group in the model.
+    :param model: nn.Module model.
+    :param lr_dict: Dictionary where keys are group names and values are the learning rates.
+    :return: List of param groups with named_parameters and corresponding learning rates.
+    """
+    param_groups = []
+    default_lr = lr_dict.get("default", None)
+    if default_lr is None:
+        raise RuntimeError("When passing initial_lr as dictionary, must pass 'default'.")
+    group_names = set(lr_dict.keys()) - {"default"}
+
+    for group_name in group_names:
+        lr = lr_dict[group_name]
+        named_params = [(name, param) for name, param in model.named_parameters() if name.startswith(group_name)]
+
+        if lr == 0:
```

Comment:
I think this check can be simplified/improved to this:

```
optimizable_params = [(name, param) for name, param in named_params if param.requires_grad and lr > 0]
param_groups.append({"named_params": optimizable_params , "lr": lr, "name": group_name})
```

Why: We have to respect original value of `requires_grad` from the model and not change it. 
Instead we just filter out what should not be optimized